app/lib/features/chat_ng/widgets/events/member_update_event.dart

```diff
@@ -31,87 +36,40 @@ class MemberUpdateEvent extends ConsumerWidget {
-    final firstName =
+    final senderName =
         ref
             .watch(
               memberDisplayNameProvider((roomId: roomId, userId: senderId)),
             )
-            .valueOrNull;
-    final msgContent = item.msgContent()?.body() ?? '';
-
-    return switch (eventType) {
-      'ProfileChange' => switch (msgType) {
-        'ChangedDisplayName' =>
-          '${lang.chatDisplayNameUpdate(firstName ?? senderId)} $msgContent',
-        'SetDisplayName' =>
-          '${lang.chatDisplayNameSet(firstName ?? senderId)}: $msgContent',
-        'RemoveDisplayName' => lang.chatDisplayNameUnset(firstName ?? senderId),
-        'ChangeProfileAvatar' => lang.chatUserAvatarChange(
-          firstName ?? senderId,
-        ),
-        _ => msgContent,
-      },
-      _ => switch (msgType) {
-        'Joined' =>
-          isMe
-              ? lang.chatYouJoined
-              : firstName != null
-              ? lang.chatJoinedDisplayName(firstName)
-              : lang.chatJoinedUserId(senderId),
-        'Left' =>
-          isMe ? lang.chatYouLeft : lang.chatUserLeft(firstName ?? senderId),
-        'Banned' =>
-          isMe
-              ? lang.chatYouBanned(msgContent)
-              : lang.chatUserBanned(firstName ?? senderId, msgContent),
-        'Unbanned' =>
-          isMe
-              ? lang.chatYouUnbanned(msgContent)
-              : lang.chatUserUnbanned(firstName ?? senderId, msgContent),
-        'Kicked' =>
-          isMe
-              ? lang.chatYouKicked(msgContent)
-              : lang.chatUserKicked(firstName ?? senderId, msgContent),
-        'KickedAndBanned' =>
-          isMe
-              ? lang.chatYouKickedBanned(msgContent)
-              : lang.chatUserKickedBanned(firstName ?? senderId, msgContent),
-        'InvitationAccepted' =>
-          isMe
-              ? lang.chatYouAcceptedInvite
-              : firstName != null
-              ? lang.chatInvitationAcceptedDisplayName(firstName)
-              : lang.chatInvitationAcceptedUserId(senderId),
-        'Invited' =>
-          (() {
-            final inviteeId = msgContent;
-            final inviteeName =
-                ref
-                    .watch(
-                      memberDisplayNameProvider((
-                        roomId: roomId,
-                        userId: inviteeId,
-                      )),
-                    )
-                    .valueOrNull;
-            return isMe
-                ? lang.chatYouInvited(inviteeName ?? inviteeId)
-                : firstName != null && inviteeName != null
-                ? lang.chatInvitedDisplayName(inviteeName, firstName)
-                : lang.chatInvitedUserId(inviteeId, senderId);
-          })(),
-        _ => msgContent,
-      },
-    };
+            .valueOrNull ??
+        simplifyUserId(senderId) ??
+        senderId;
+    MembershipContent content = item.membershipContent().expect(
+      'failed to get content of membership change',
+    );
```

Comment:
don't expect in build functions.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2790/commits/69e0f58102392c3f9065aae26961522382532bbe